### PR TITLE
feat: shift recommendations experiment to optimizely full stack

### PR DESCRIPTION
**Pull request by syedsajjadkazmii**
VAN-1330

### Description

This PR converts post-registration recommendations optimizely web experiment to optimizely full stack experiment.

#### How Has This Been Tested?

Tested locally with optimizely full stack logs

### CODE REVIEW

File: src/recommendations/optimizelyExperiment.js [hunk 1]
```diff
@@ -1,21 +1,47 @@
-const RECOMMENDATIONS_EXP_ID = process.env.RECOMMENDATIONS_EXPERIMENT_ID;
-const RECOMMENDATIONS_EXP_VARIATION = 'show_recommendations_page';
+import optimizelyInstance from '../data/optimizely';
```
[the remark this answers is not in the document]
Comment (syedsajjadkazmii) [reply src/recommendations/optimizelyExperiment.js:1]: yes, default export can be imported with any name

File: src/progressive-profiling/ProgressiveProfiling.jsx [hunk 2]
```diff
@@ -67,22 +70,18 @@ const ProgressiveProfiling = (props) => {
...
-          setShowRecommendationsPage(showRecommendations);
-          if (!showRecommendations) {
-            trackRecommendationsViewed([], true, authenticatedUser?.userId);
-          }
-        }, 500);
+        const userIdStr = authenticatedUser?.userId.toString();
+        const showRecommendations = activateRecommendationsExperiment(userIdStr);
+        setShowRecommendationsPage(showRecommendations);
+        if (!showRecommendations) {
+          trackRecommendationsViewedOptimizely(userIdStr);
```
[the remark this answers is not in the document]
Comment (syedsajjadkazmii) [reply src/progressive-profiling/ProgressiveProfiling.jsx:80]: we want to fire recommendation viewed event for both variation and original group

File: src/recommendations/optimizelyExperiment.js [hunk 3]
```diff
@@ -1,21 +1,47 @@
...
+export const eventNames = {
+  recommendedCourseClicked: 'welcome_page_recommendation_card_click',
+  recommendationsViewed: 'welcome_page_recommendations_viewed',
+};
+
+/**
+  * Activate the post registration recommendations optimizely experiment
+  * and return the true if the user is in variation else false.
+  * @param  {String} userId  user id of authenticated user.
+  * @return {boolean}        true if the user is in variation else false
```
[the remark this answers is not in the document]
Comment (syedsajjadkazmii) [reply src/recommendations/optimizelyExperiment.js:15]: sure

> Can we not use `window` object to load optimizely instead to optimizely npm?

As per my knowledge, we can integrate optimizely full-stack in our app using SDKs only.

**Comment by syedsajjadkazmii** [comment]
> Can we not use `window` object to load optimizely instead to optimizely npm?

Can we not use window object to load optimizely instead to optimizely npm?

As per my knowledge, we can integrate optimizely full-stack in our app using SDKs only.

**Review by mubbsharanwar — APPROVED** [review]

**Pull request merged**